mesop/server/static_file_serving.py

```diff
@@ -159,7 +159,7 @@ def add_security_headers(response: Response):
         "style-src": "'self' 'unsafe-inline' fonts.googleapis.com",
         "script-src": f"'self' 'nonce-{g.csp_nonce}'",
         # https://angular.io/guide/security#enforcing-trusted-types
-        "trusted-types": "angular angular#unsafe-bypass lit-html",
+        "trusted-types": "angular angular#unsafe-bypass lit-html 'allow-duplicates'",
```

Comment:
This is one issue I ran into. For some reason lit-html was getting added twice I think because I needed to also import unsafeHTML. And so I added allow-duplicates. Not sure if that's the best idea or not.

Reply:
I think we can avoid the trusted types duplicates issue by importing Lit from the exact same URL across all our web components (which is a good thing in general, to avoid loading Lit multiple times).
Right now we import Lit from 'https://cdn.jsdelivr.net/gh/lit/dist@3/core/lit-core.min.js'

In this PR, you're importing Lit from 'https://cdn.jsdelivr.net/npm/lit@3.1.4/+esm';

I think you could try swapping out the esm URL for the first URL which we already use and then you won't load Lit twice. I'm not sure if you had to use the +esm directive.

Alternatively, you could swap out **all** the usages of  'https://cdn.jsdelivr.net/gh/lit/dist@3/core/lit-core.min.js'  with 'https://cdn.jsdelivr.net/gh/lit/dist@3/all/lit-all.min.js' which should include the unsafe directive you need.

BTW, these are the bundle URLs recommended by the Lit docs: https://lit.dev/docs/getting-started/#use-bundles - I'm not sure what the differences are w/ the esm one, but the regular ones (from the docs) seems to work fine with ES modules.

Reply:
Yeah, this is one of the pain points of working with the web components in Mesop. I feel like I spend most of my time trying to figure out how to get the imports to work.

I think the issue with `https://cdn.jsdelivr.net/gh/lit/dist@3/core/lit-core.min.js` this URL is that it does not include unsafeHTML, which I needed in this case. So the only alternative would be to use the `lit-all.min.js` I think. I did try to include the unsafe HTML directive directly along with lit-core.min.js, but it didn't seem to find it.

Which is how I ended up with using the esm modules. Also when I search jsdelivr, the gh URL does not show up in the results.

However, I don't know if this import https://cdn.jsdelivr.net/npm/lit@3.1.4/+esm, is the same as lit-core.min.js in terms of what it includes.

Also I think when I import https://cdn.jsdelivr.net/npm/lit@3.1.4/+esm, it does not trigger the duplicate CSP issue for lit-html. iirc only does it when I include the unsafe-html directive.

--------

But I'll give give it another try.

Reply:
In the end, it seems I needed to set `dangerously_disable_trusted_types=True` for DomPurifier so the duplicate lit-html didn't matter anymore. But I'll keep an eye on this when I add more web components.